plugin/pkg/admission/runtimeclass/admission.go

```diff
@@ -0,0 +1,199 @@
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return NewRuntimeClass(), nil
+	})
+}
+
+// RuntimeClass is an implementation of admission.Interface.
+// It looks at all new pods and sets pod.Spec.Overhead if a RuntimeClass is specified which
+// defines an Overhead. TODO: add blurb about validation/rejection handling
+type RuntimeClass struct {
+	*admission.Handler
+	client kubernetes.Interface
+	lister nodev1beta1listers.RuntimeClassLister
+}
+
+var _ admission.MutationInterface = &RuntimeClass{}
+var _ admission.ValidationInterface = &RuntimeClass{}
+
+var _ genericadmissioninitailizer.WantsExternalKubeInformerFactory = &RuntimeClass{}
+var _ genericadmissioninitailizer.WantsExternalKubeClientSet = &RuntimeClass{}
+
+func (r *RuntimeClass) SetExternalKubeInformerFactory(f informers.SharedInformerFactory) {
+	runtimeClassInformer := f.Node().V1beta1().RuntimeClasses()
+	r.SetReadyFunc(runtimeClassInformer.Informer().HasSynced)
+	r.lister = runtimeClassInformer.Lister()
+}
+
+func (r *RuntimeClass) SetExternalKubeClientSet(client kubernetes.Interface) {
+	r.client = client
+}
+
+func (r *RuntimeClass) ValidateInitialization() error {
+	if r.lister == nil {
+		return fmt.Errorf("missing RuntimeClass lister")
+	}
+	if r.client == nil {
+		return fmt.Errorf("missing client")
+	}
+	return nil
+}
+
+// Admit makes an admission decision based on the request attributes
+func (r *RuntimeClass) Admit(attributes admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	return admissionAction(r, attributes, "admit")
+}
+
+// Validate makes sure that pod adhere's to RuntimeClass's definition
+func (r *RuntimeClass) Validate(attributes admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	return admissionAction(r, attributes, "validate")
+}
+
+// NewRuntimeClass creates a new always pull images admission control handler
+func NewRuntimeClass() *RuntimeClass {
+	return &RuntimeClass{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+	}
+}
+
+// admissionAction handles Admit and Validate phases of admission, switching based on the admissionPhase parameter
+func admissionAction(r *RuntimeClass, attributes admission.Attributes, admissionPhase string) (err error) {
+
+	// Ignore all calls to subresources or resources other than pods.
+	if shouldIgnore(attributes) {
+		return nil
+	}
+	pod, ok := attributes.GetObject().(*api.Pod)
+	if !ok {
+		return apierrors.NewBadRequest("Resource was marked with kind Pod but was unable to be converted")
+	}
+
+	// Get RuntimeClass. Not finding a given runtimeClass does not return an error. We will want to continue in the
+	// case of Validate phase, even if a RuntimeClass is not defined
+	runtimeClass, err := getRuntimeClass(r, pod, pod.Spec.RuntimeClassName)
+	if err != nil {
+		return err
+	}
+
+	switch admissionPhase {
+	case "admit":
+		err := setOverhead(attributes, pod, runtimeClass)
```

Comment:
The overhead operations should be behind the feature gate, since we'll reuse this controller for scheduling too.

Reply:
done.